loginserver/world_server.cpp

```diff
@@ -1348,3 +1354,36 @@ void WorldServer::OnKeepAlive(EQ::Timer *t)
 	ServerPacket pack(ServerOP_KeepAlive, 0);
 	m_connection->SendPacket(&pack);
 }
+
+void WorldServer::FormatWorldServerName(char *name, int8 server_list_type)
+{
+	std::string server_long_name = name;
+	server_long_name = trim(server_long_name);
+
+	bool name_set_to_bottom = false;
+	if (server.options.IsWorldDevTestServersListBottom() && server_list_type == LS::ServerType::Standard) {
+		std::string s = str_tolower(server_long_name);
+		if (s.find("dev") != std::string::npos) {
+			server_long_name   = fmt::format("|D| {}", server_long_name);
+			name_set_to_bottom = true;
+		}
+		else if (s.find("test") != std::string::npos) {
+			server_long_name   = fmt::format("|T| {}", server_long_name);
+			name_set_to_bottom = true;
+		}
+		else if (s.find("installer") != std::string::npos) {
+			server_long_name   = fmt::format("|I| {}", server_long_name);
+			name_set_to_bottom = true;
+		}
+	}
+	if (server.options.IsWorldSpecialCharacterStartListBottom() && server_list_type == LS::ServerType::Standard &&
+		!name_set_to_bottom) {
+		auto first_char = server_long_name.c_str()[0];
+		if (IsAllowedWorldServerCharacterList(first_char) && first_char != '|') {
+			server_long_name   = fmt::format("|*| {}", server_long_name);
+			name_set_to_bottom = true;
+		}
+	}
+
+	strcpy(name, server_long_name.c_str());
```

Comment:
This could be unsafe if a server long name is right at the max buffer length since this function can grow the string. May want to pass the max buffer length to the function as another arg and use `strn0cpy(name, server_long_name.c_str(), length)`

Reply:
Good call and good eye; changed